### PR TITLE
github: add rust action

### PR DESCRIPTION
**Pull request by vbatts**
validate PRs with a github action to check build and test of the rust code

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Rust
```
Comment (invidian) [inline .github/workflows/rust.yml]: Do we have other language to test here? If it runs only for PRs, can we name it `CI` maybe?

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Build
+      run: cargo build --verbose
```
Comment (invidian) [inline .github/workflows/rust.yml:20]: Is verbose needed here?

File: .github/workflows/rust.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    - name: Run tests
+      run: cargo test --verbose
```
Comment (invidian) [inline .github/workflows/rust.yml:22]: Ditto verbose? With equivalent of `go test -v`, you should only use `-v` while developing, so it's easy to find what failed.

File: .github/workflows/rust.yml (hunk 1 above)
Comment (vbatts) [reply .github/workflows/rust.yml]: fair. This and the two comments you made below are all default from Github's rust action. I made no modifications.

File: .github/workflows/rust.yml (hunk 2 above)
Comment (vbatts) [reply .github/workflows/rust.yml:20]: ditto

File: .github/workflows/rust.yml (hunk 3 above)
Comment (vbatts) [reply .github/workflows/rust.yml:22]: ditto

File: .github/workflows/rust.yml (hunk 1 above)
Comment (invidian) [reply .github/workflows/rust.yml]: I guess it's fine then? :smile:

Comment (vbatts) [reply .github/workflows/rust.yml]: i split the difference and called it "Rust CI" :joy_cat:

**Review by dongsupark — APPROVED** [review]

**Pull request merged**